plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -0,0 +1,116 @@
+package com.google.cloud.tools.eclipse.usagetracker;
+
+import com.google.api.client.util.escape.PercentEscaper;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+
+import org.eclipse.core.runtime.Platform;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Provides methods that report plugin-specific events to Analytics.
+ */
+public class AnalyticsPingManager {
+
+  private static final String ANALYTICS_COLLECTION_URL = "https://ssl.google-analytics.com/collect";
+  // TODO: retrieve tracking ID from a system environment variable.
+  private static final String ANALYTICS_TRACKING_ID = "@ga.tracking.id@";
+  private static final String APPLICATION_NAME = "gcloud-eclipse-tools";
+
+  // TODO: see if there a way to get a unique ID for each Eclipse installation.
+  private static final String CLIENT_ID = UUID.randomUUID().toString();
+
+  // Fixed-value query parameters present in every ping, and their fixed values:
+  //
+  // For the semantics of each parameter, consult the followings:
+  //
+  // https://github.com/google/cloud-reporting/blob/master/src/main/java/com/google/cloud/metrics/MetricsUtils.java#L183
+  // https://developers.google.com/analytics/devguides/collection/protocol/v1/reference
+  private static final ImmutableMap<String, String> STANDARD_PARAMETERS =
```

Comment:
Not sure what's really going on, but I added `@SuppressWarnings("serial") suggested by Eclipse. Seems related to the inner anonymous map(?).